File: poker_arena_server/src/utils/Logger.ts
/**
 * 简单的日志工具
 * 支持不同日志级别，可通过配置控制输出
 */

export enum LogLevel {
    DEBUG = 0,
    INFO = 1,
    WARN = 2,
    ERROR = 3,
    NONE = 4
}

const LOG_LEVEL_NAMES: Record<LogLevel, string> = {
    [LogLevel.DEBUG]: 'DEBUG',
    [LogLevel.INFO]: 'INFO',
    [LogLevel.WARN]: 'WARN',
    [LogLevel.ERROR]: 'ERROR',
    [LogLevel.NONE]: 'NONE'
};

class LoggerClass {
    private level: LogLevel = LogLevel.INFO;

    /**
     * 设置日志级别
     */
    public setLevel(level: LogLevel | string): void {
        if (typeof level === 'string') {
            const upperLevel = level.toUpperCase();
            switch (upperLevel) {
                case 'DEBUG': this.level = LogLevel.DEBUG; break;
                case 'INFO': this.level = LogLevel.INFO; break;
                case 'WARN': this.level = LogLevel.WARN; break;
                case 'ERROR': this.level = LogLevel.ERROR; break;
                case 'NONE': this.level = LogLevel.NONE; break;
                default: this.level = LogLevel.INFO;
            }
        } else {
            this.level = level;
        }
    }

    /**
     * 获取当前日志级别
     */
    public getLevel(): LogLevel {
        return this.level;
    }

    /**
     * DEBUG 级别日志
     */
    public debug(tag: string, message: string, ...args: unknown[]): void {
        if (this.level <= LogLevel.DEBUG) {
            console.log(`[${tag}] ${message}`, ...args);
        }
    }

    /**
     * INFO 级别日志
     */
    public info(tag: string, message: string, ...args: unknown[]): void {
        if (this.level <= LogLevel.INFO) {
            console.log(`[${tag}] ${message}`, ...args);
        }
    }

    /**
     * WARN 级别日志
     */
    public warn(tag: string, message: string, ...args: unknown[]): void {
        if (this.level <= LogLevel.WARN) {
            console.warn(`[${tag}] ${message}`, ...args);
        }
    }

    /**
     * ERROR 级别日志
     */
    public error(tag: string, message: string, ...args: unknown[]): void {
        if (this.level <= LogLevel.ERROR) {
            console.error(`[${tag}] ${message}`, ...args);
        }
    }
}

// 单例导出
export const Logger = new LoggerClass();

File: poker_arena_server_go/util/logger.go
package util

import (
	"fmt"
	"log"
	"os"
	"strings"
)

type LogLevel int

const (
	LevelDebug LogLevel = iota
	LevelInfo
	LevelWarn
	LevelError
	LevelNone
)

var logger = &Logger{level: LevelInfo}

type Logger struct {
	level LogLevel
}

func SetLogLevel(level string) {
	switch strings.ToUpper(level) {
	case "DEBUG":
		logger.level = LevelDebug
	case "INFO":
		logger.level = LevelInfo
	case "WARN":
		logger.level = LevelWarn
	case "ERROR":
		logger.level = LevelError
	case "NONE":
		logger.level = LevelNone
	default:
		logger.level = LevelInfo
	}
}

func Debug(tag, msg string, args ...interface{}) {
	if logger.level <= LevelDebug {
		log.Printf("[%s] %s", tag, fmt.Sprintf(msg, args...))
	}
}

func Info(tag, msg string, args ...interface{}) {
	if logger.level <= LevelInfo {
		log.Printf("[%s] %s", tag, fmt.Sprintf(msg, args...))
	}
}

func Warn(tag, msg string, args ...interface{}) {
	if logger.level <= LevelWarn {
		log.Printf("[%s] WARN: %s", tag, fmt.Sprintf(msg, args...))
	}
}

func Error(tag, msg string, args ...interface{}) {
	if logger.level <= LevelError {
		log.Printf("[%s] ERROR: %s", tag, fmt.Sprintf(msg, args...))
	}
}

func Fatal(tag, msg string, args ...interface{}) {
	log.Printf("[%s] FATAL: %s", tag, fmt.Sprintf(msg, args...))
	os.Exit(1)
}
